pkg/kubelet/kubelet.go

```diff
@@ -1541,7 +1541,10 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	existingStatus, ok := kl.statusManager.GetPodStatus(pod.UID)
 	if !ok || existingStatus.Phase == v1.PodPending && apiPodStatus.Phase == v1.PodRunning &&
 		!firstSeenTime.IsZero() {
-		metrics.PodStartDuration.Observe(metrics.SinceInSeconds(firstSeenTime))
+		if !pod.CreationTimestamp.IsZero() {
+			metrics.PodStartE2EDuration.WithLabelValues(pod.Namespace).Observe(metrics.SinceInSeconds(pod.CreationTimestamp.Time))
```

Comment:
This may be slightly confusing for users, since the kubelet hasn't necessarily updated the pod status to running at this point.  The kubelet status manager has a background process to perform updates to pods, but can often be limited by QPS limits for scalability reasons.  It often takes ~30 seconds for the kubelet to update status after actual containers are running.  We may want to consider only recording this metric in the status manger after the status update has been sent to the API Server.

Reply:
I added metrics here simply because pod start duration is already added here. to me it's more of a matter of fact if we want to record the time where kubelet starts to do the work (after which user workload could have already be running), or the time where kublet has notified user about it's status (i.e. update pod status).

thoughts? @logicalhan